src/components/mdxComponents.tsx

```diff
@@ -39,41 +39,44 @@ export default {
             '& + *': {
               marginTop: '8px',
             },
-            '& a': {
-              position: 'absolute',
-              inset: 0,
-              '& .icon': {
-                height: '100vh',
-                display: 'none',
-                _after: {
-                  position: 'absolute',
-                  content: '"#"',
-                  left: '-1em',
-                  color: '#0d9488',
-                },
-              },
-              _hover: {
-                '& .icon': {
-                  display: 'inline',
-                },
-              },
-            },
           })
         )}
       >
         <Slot />
       </h1>
     );
   }),
-  h2: component$(() => (
+  h2: component$((props) => (
     <h2
+      {...props}
```

Comment:
Used for the rehypeAutolinkHeadings vite plugin to automatically pass the id for navigation.